### PR TITLE
Add workflow for running integration tests on a dedicated VM

### PR DESCRIPTION
**Pull request by RomanIakovlev**
This change adds a new workflow to run the existing integration tests on a dedicated virtual machine, created and destroyed for each test run. 

This workflow creates and provisions a VM, using the Docker Compose setup available at https://github.com/clearlydefined/docker_dev_env_experiment. After provisioning, it runs the existing integration test suite against that newly created environment.

Advantage of running tests on custom infra is that it's possible to use a more powerful machine for the crawler, leading to a much faster test execution. E.g. static components test completes in ~5 minutes on a 16-core VM.

### CODE REVIEW

**Review by qtomlinson — COMMENTED** [review]
Great improvement for running the integration tests!  Minor comments for the tests.

**Comment by qtomlinson** [comment]
It is good that the original integration tests on dev is preserved because the dev deployment set up is the closest to production setup.  The interaction with queues and blob store and trimmed definition collection is in dev deployment, which are not tested here.

File: .github/workflows/integration-test-custom.yml [hunk 1]
```diff
@@ -0,0 +1,260 @@
...
+              fi
+            done
+            
+            # Copy sample_env to .env and replace tokens
+            cp sample_env .env
+            
+            # Replace both GitHub tokens with the provided token
+            sed -i "s/^CRAWLER_GITHUB_TOKEN=.*/CRAWLER_GITHUB_TOKEN=${GITHUB_TOKEN}/" .env
+            sed -i "s/^CURATION_GITHUB_TOKEN=.*/CURATION_GITHUB_TOKEN=${GITHUB_TOKEN}/" .env
+            echo "" >> .env && echo "CRAWLER_SCANCODE_PARALLELISM=14" >> .env
```
Comment (qtomlinson) [inline .github/workflows/integration-test-custom.yml:185]: AS a separate PR, could we document this in https://github.com/clearlydefined/operations/blob/main/overview.md?

**Comment by RomanIakovlev** [comment]
@qtomlinson thanks for your feedback, I've added the changes you requested. I have to admit, it looks much better this way!

**Review by qtomlinson — APPROVED** [review]
LGTM!

**Pull request merged**